tests/common.go

```diff
@@ -5311,3 +5318,116 @@ func IsEksPxOperator() bool {
 	}
 	return false
 }
+
+// ConcurrentExecute executes the given operation on each item in the collection, either sequentially
+// or in parallel, depending on the specified execution mode. It also returns an error when items is not
+// of type slice, array or map.
+//
+// Parameters:
+//
+//	items: The collection of items to operate on (slice, array, or map).
+//	operation: The function to execute on each item. If the function takes one argument,
+//	           it will be passed the item value. If it takes two arguments, the first
+//	           will be the item key or index, and the second will be the item value.
+//	executionMode: The mode to use for executing the operation, either "Sequential" or "Parallel".
+//
+// # Example
+//
+// The original code:
```

Comment:
@lsrinivas-pure would it be okay if we add this part?

Reply:
ya it is fine, only thing Im thinking is about logging. 
do we need to log about context also like which ctx it is validating when it is running parallely